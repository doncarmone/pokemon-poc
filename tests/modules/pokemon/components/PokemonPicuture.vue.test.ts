import { mount } from "@vue/test-utils";
import PokemonPicture from '@pokemon/components/PokemonPicture.vue';

describe('PokemonPicture', () => {
    test('should render hidden image when showPokemon prop is false', async () => {
        const wrapper = mount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false

            }
        })
        expect(wrapper.find('.brightness-0').exists()).toBe(true)
    })

    test('should render hidden image when showPokemon prop is true', async () => {
        const wrapper = mount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: true

            }
        })
        expect(wrapper.find('.fade-in').exists()).toBe(true)
    })
})
