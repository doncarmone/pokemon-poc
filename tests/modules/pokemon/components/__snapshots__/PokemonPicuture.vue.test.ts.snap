// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonPicture > should render hidden image when showPokemon prop is true 1`] = `"<section data-v-49cc19b7=""><img data-v-49cc19b7="" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png" class="fade-in"></section>"`;
